src/hls-adapter.js

```diff
@@ -1101,6 +1099,11 @@ export default class HlsAdapter extends BaseMediaSourceAdapter {
     const errorFatal = data.fatal;
     let errorDataObject = this._getErrorDataObject(data);
     if (errorFatal) {
+      if (this._loadPromiseHandlers) {
+        const {type, details, url} = data;
+        this._loadPromiseHandlers.reject({type, details, url});
```

Comment:
reject after it has been resolved? sounds incorrect

Reply:
do you mean which is NOT fired in case of an error?

Reply:
Right fixed